src/main/java/teammates/storage/api/NotificationsDb.java

```diff
@@ -37,6 +42,52 @@ public NotificationAttributes getNotification(String notificationId) {
         return notificationId.isEmpty() ? null : makeAttributesOrNull(getNotificationEntity(notificationId));
     }
 
+    /**
+     * Gets notifications by {@code targetUser}.
+     *
+     * @return a list of notifications for the specified targetUser.
+     */
+    public List<NotificationAttributes> getNotificationsByTargetUser(NotificationTargetUser targetUser) {
+        assert targetUser != null;
+
+        List<Notification> endEntities = load()
```

Comment:
Would we be able to construct an OR filter so that we can filter out targetUser OR "general" at the same time. Not sure if its necessary to do 2 filters and if it will affect performance

Something I saw on [composite filters](https://cloud.google.com/appengine/docs/standard/java/datastore/queries)

Reply:
After searching and trials, I don't think the SDK currently supports `IN` filters. Or if anyone knows a way, please let me know while I search more.